GSEAPY
========

GSEAPY is a python wrapper for **GESA**. 

`Gene Set Enrichment Analysis <http://software.broadinstitute.org/gsea/index.jsp>`_ (GSEA) 
is a computational method that determines whether an a priori defined set of genes shows 
statistically significant, concordant differences between two biological states (e.g. phenotypes). 

The full ``GSEA`` is far too extensive to describe here; see
`GSEA  <http://www.broadinstitute.org/cancer/software/gsea/wiki/index.php/Main_Page>`_ documentation for more information.


However, the R version of GSEA had not been updated since 2006. 
Although the best choice of GSEA is the desktop Java version, this version did not provide means to 
modify plots, like legends, ticks......


GSEAPY is  used for convenient GO enrichments and produce **publishable quality figures** from python. 

GSEAPY could be used for **RNA-seq, ChIP-seq, Microarry** data.



**Features of GSEAPY**

#. GSEAPY will reproduce the GSEA figures using GSEA desktop version results.

#. Statistical testing module is identical to ``GSEA`` Desktop version.

#. Enhancement of ``GSEA`` will be considered. If you would like to contribute, please @BioNinja on ``Github``. 



GSEA desktop version output: 
-------------------------------------------------
This is an example of GSEA desktop application output

.. figure:: GSEA_OCT4_KD.png

    Generated by ``GSEA`` desktop version


GSEAPY output
-----------------------------------------------
Using the same algorithm by ``GSEA``, GSEAPY reproduce the example above.

.. figure:: gseapy_OCT4_KD.png

   
   
   Generated by GSEAPY
   
   **GSEAPY figures are PDF formats by default.** Other matplotlib figures formats are supported, too.

   You can modify ``GSEA`` plots easily in .pdf files. Please Enjoy.



Installation
------------

| Install gseapy package from pypi and download 

.. code:: shell

   $ pip install gseapy

| You may instead want to use the development version from Github, by running

.. code:: shell

   $ pip install git+git://github.com/BioNinja/gseapy.git#egg=gseapy

Dependency
--------------
* Python 2.7 or 3.3+

Mandatory
~~~~~~~~~

* Numpy 
* Pandas 
* Matplotlib
* Beautifulsoup4

You may also need **lxml, html5lib**, if you could not parse xml files. 


   
Run GSEAPY
-----------------

| Gseapy has two subcommands: ``replot``, ``call``.
| ``replot`` command will parse the GSEA desktop results directory(the location of edb file folder). 
| ``call`` command will compute es,nes,pval,fdr and generates plots, *de novo*.


| For more details,see below. 


For command line usage:
~~~~~~~~~~~~~~~~~~~~~~~

The ``replot`` module reproduce GSEA desktop version results. The only input of GSEAPY is the location to GSEA results.
The ``call`` module reproduce GSEAPY results. The input requries a txt file( FPKM, Expected Counts, TPM, et.al), an cls file,
and gene_sets file in gmt format.


.. code:: shell
  
  $ gseapy replot -i path/to/gsea_edb_file_folder_location -o gesapy_out

  # example
  $ gseapy replot -i ./Gsea.reports -o test



Run gseapy inside python:
~~~~~~~~~~~~~~~~~~~~~~~~~~~

.. code:: python
  
   import gseapy
   # reproduce plots from GSEA ouput
   gseapy.replot('./Gsea.reports','test')

   # calculate es, nes, pval,fdrs, and produce figures using gseapy.
   gseapy.run(data = expression.txt, gene_sets = gene_sets.gmt,cls = test.cls, 
             min_size = 15, max_size = 1000, permutation_n = 1000, weighted_score_type = 1,
             permutation_type = 'gene_set', method = 'log2_ratio_of_classes',ascending = False, 
             outdir='test',figsize = (6.5,6),format = 'png')



To see GSEAPY sub-commands.
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.. code:: shell
   
   $ gseapy --help 
   
    usage: gseapy [-h] [--version] {call,replot} ...

    gseapy -- GSEAPY for Gene Set Enrichment Analysis

    positional arguments:
      {call,replot}
        call       Main GSEAPY Function: run GSEAPY instead of GSEA.
        replot     Reproduce GSEA desktop figures.

    optional arguments:
      -h, --help   show this help message and exit
      --version    show program's version number and exit



For command line options of each command, type: gseapy COMMAND -h


The ``replot`` module
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

.. code:: shell

   $ gseapy replot -h

   usage: gseapy replot [-h] -i [-o] [-f] [--figsize width height] [-w]

    optional arguments:
      -h, --help            show this help message and exit

    Positional arguments:
      -i , --indir          The GSEA desktop results directroy that you want to
                            reproduce the figure
      -o , --outdir         The gseapy output directory. Default: the current
                            working directory
      -f , --format         Format of output figures, choose from {'pdf', 'png',
                            'jpeg', 'eps'}. Default: 'pdf'.
      --figsize width height
                            The figsize keyword argument need two parameter to
                            define. Default: [6.5, 6]
      -w , --weight         Weighted_score type of rank_metrics.Choose from (0, 1,
                            1.5, 2),default: 1



The ``call`` module
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

.. code:: shell

   $ gseapy call -h

    usage: gseapy call [-h] -i DATA -c CLS -g GMT [-p] [-o] [-f]
                        [--figsize width height] [--min-size] [--max-size] [-n]
                        [-w] [-m] [-a]
    
    optional arguments:
      -h, --help            show this help message and exit
        
    Input files arguments:
      -d DATA, --datab DATA
                            Expression table of phenotypes. Expected a txt
                            file.Same with GSEA.
      -c CLS, --cls CLS     .cls files. Same with GSEA.
      -g GMT, --gmt GMT     Gene Sets in gmt format. Same with GSEA.
      -p , --permu-type
                            Gene Sets in gmt format. Same with GSEA, choose from
                            {'gene_set', 'phenotype'}
    
    Output arguments:
      -o , --outdir         The gseapy output directory. Default: the current
                            working directory
      -f , --format         Format of output figures, choose from {'pdf', 'png',
                            'jpeg', 'eps'}. Default: 'pdf'.
      --figsize width height
                            The figsize keyword argument need two parameter to
                            define. Default: [6.5,6]
    
    GSEA advance arguments:
      --min-size            Min size of gene sets. Default: 15
      --max-size            Max size of gene sets. Default: 1000
      -n , --permu-num      Permutation number. Default: 1000
      -w , --weight         Weighted_score type of rank_metrics.Choose from {0, 1,
                            1.5, 2},default: 1
      -m , --method         Methods to calculate correlations of ranking metrics.
                            Choose from {'signal_to_noise', 't_test',
                            'ratio_of_classes',
                            'diff_of_classes','log2_ratio_of_classes'}. Default:
                            'log2_ratio_of_classes'
      -a, --ascending       Rank metrice sorting order. If the -a flag was chosen,
                            then ascending equals to True. Default: False.



    







   
